frontend/background.js

```diff
@@ -1,12 +1,33 @@
 console.log("background.js is executed when extension is installed or refreshed");
 
-const listOfInjectableURLRegex = [ // change these to the list of allowed news sources
-    "^https:\\/\\/www\\.google",
-    "^https:\\/\\/www\\.youtube\\.com",
-    "^https:\\/\\/www\\.stackoverflow\\.com",
-    "^https:\\/\\/www\\.bbc\\.com\\/news\\/"
+var listOfInjectableURLRegex = [ // change these to the list of allowed news sources
+    // "^https:\\/\\/www\\.google",
+    // "^https:\\/\\/www\\.youtube\\.com",
+    // "^https:\\/\\/www\\.stackoverflow\\.com",
+    // "^https:\\/\\/www\\.bbc\\.com\\/news\\/"
 ]
 
+// Parses allowed news sources from news-sources.json
+function parseAllowedNewsSource() {
+    // Fetches data from local .json file
+    fetch('../static/news-sources.json')
+        .then(response => response.json())
+        .then(json => addToListOfInjectableURLRegex(json))
+
+}
+
+// Helper function for parseAllowedNewsSource
+function addToListOfInjectableURLRegex(json) {
+    const size = json.numRows;
+    for (i = 0; i < size; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < size; i++) {
```

I would just use `forEach` though to avoid the indexing